src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -63,6 +63,23 @@ public ReferenceInfo(ConfigurationMetadata referenceConfiguration, ProjectItemIn
             }
         }
 
+        private readonly struct TargetSpecification
+        {
+            public TargetSpecification(string target, bool skipIfNonexistent)
+            {
+                ErrorUtilities.VerifyThrow(
+                    !skipIfNonexistent || (!target.Equals(MSBuildConstants.DefaultTargetsMarker)
```

Comment:
NIT: opinionated, but I believe that avoiding NOTs will make it more readable:
```
!(skipIfNonexistent 
  && (target.Equals(MSBuildConstants.DefaultTargetsMarker 
   || target.Equals(MSBuildConstants.ProjectReferenceTargetsOrDefaultTargetsMarker))
````
As we are asserting exceptional conditions which shall not happen

Reply:
I added an explanatory comment to enhance readability.